okhttp/src/main/kotlin/okhttp3/internal/connection/RealConnection.kt

```diff
@@ -705,43 +699,38 @@ class RealConnection(
    * Track a failure using this connection. This may prevent both the connection and its route from
    * being used for future exchanges.
    */
-  internal fun trackFailure(call: RealCall, e: IOException?) {
-    assertThreadDoesntHoldLock()
-
-    synchronized(this) {
-      if (e is StreamResetException) {
-        when {
-          e.errorCode == ErrorCode.REFUSED_STREAM -> {
-            // Stop using this connection on the 2nd REFUSED_STREAM error.
-            refusedStreamCount++
-            if (refusedStreamCount > 1) {
-              noNewExchanges = true
-              routeFailureCount++
-            }
-          }
-
-          e.errorCode == ErrorCode.CANCEL && call.isCanceled() -> {
-            // Permit any number of CANCEL errors on locally-canceled calls.
-          }
-
-          else -> {
-            // Everything else wants a fresh connection.
+  @Synchronized internal fun trackFailure(call: RealCall, e: IOException?) {
```

Comment:
this diff reads better hiding whitespace changes. I’m just promoting `synchronized` to `@Synchronized`.